### PR TITLE
Clear prepared statement handle before reconnect

### PR DESCRIPTION
**Pull request by lukasaignerrsg**
This fix clears the SQLServerPreparedStatement.prepStmtHandle before reconnecting.
The provided interface can also be used for other objects that need to handle reconnecting.

One test fails: PropertyTest.testRetryCount:54->testInvalidPropertyOverBrokenConnection:46 Unexpected error message occurred!Eine bestehende Verbindung wurde softwaregesteuert
durch den Hostcomputer abgebrochen
This might be due to the system being German. The test also failed before the fix.

### CODE REVIEW

**Comment by lukasaignerrsg** [comment]
> @lukasaignerrsg please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Raiffeisen Software GmbH"

**Comment by lilgreenbird** [comment]
hi @lukasaignerrsg 

thank you for submitting this I have marked this for review

**Review by tkyc — DISMISSED** [review]

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 1]
```diff
@@ -3583,4 +3594,12 @@ public void addBatch(String sql) throws SQLServerException {
         Object[] msgArgs = {"addBatch()"};
         throw new SQLServerException(this, form.format(msgArgs), null, 0, false);
     }
+
+    private void clearPrepStmtHandle(){
+        prepStmtHandle = 0;
+        cachedPreparedStatementHandle = null;
+        if(getStatementLogger().isLoggable(Level.FINER)){
```
Comment (barryw-mssql) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java:3601]: Minor formatting issue. Should have a space preceeding the initila if statement brace and after the closing brace

Comment (Jeffery-Wasty) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java:3601]:
```suggestion
        if (getStatementLogger().isLoggable(Level.FINER)){
```

Comment (lilgreenbird) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java:3601]: pls run formatter on all files

**Review by lilgreenbird — APPROVED** [review]

**Comment by lilgreenbird** [comment]
ignore failing pipelines it's an issue in our test lab

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java (hunk 1 above)
Comment (Jeffery-Wasty) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java:3601]:
```suggestion
        if (getStatementLogger().isLoggable(Level.FINER)){
```

**Review by Jeffery-Wasty — CHANGES_REQUESTED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]

**Comment by kadirchisty** [comment]
Hi @Jeffery-Wasty , We see the issue (2362) occurring intermittently in 12.4.2.jre11 version. Please let us if this fix is available in 12.8.1.jre11?

**Comment by Jeffery-Wasty** [comment]
Hi @kadirchisty,

The fix is available in the latest stable version 12.8.1.

**Pull request merged**